lib/tests/test_domainwatcher.cpp

```diff
@@ -386,22 +386,22 @@ TEST_CASE("DomainWatcher triggers WRITER callback on access file modifications",
 {
     // Ensure a WRITER‐type watcher actually fires when the .access file changes
     char const* homeDir = std::getenv("HOME");
-    REQUIRE(homeDir);
+    REQUIRE(homeDir != nullptr);
 
     auto domainPath = std::filesystem::path{homeDir} / "mxl_test_writer_event";
-    std::filesystem::remove_all(domainPath);
-    std::filesystem::create_directories(domainPath);
+    remove_all(domainPath);
+    create_directories(domainPath);
 
     // Synchronization primitives for the callback
     std::mutex mtx;
     std::condition_variable cv;
     bool notified = false;
     uuids::uuid cbId;
-    WatcherType cbType;
+    mxl::lib::WatcherType cbType;
```

Comment:
Please use `auto` + right hand side initialization (with curly braces) for these.